lib/pbench/cli/server/report.py

```diff
@@ -286,11 +290,29 @@ def report_states():
         operations[row[1]][row[2]] += 1
         if row[2] == "FAILED":
             detailer.write(f"{row[1]} {row[2]} {row[0]} {row[3]!r}")
+            if row[1] == "INDEX":
+                match = index_pattern.match(row[3])
+                if match:
+                    try:
+                        code = int(match.group(1))
+                        message = match.group(2)
+                        index_errors[code] += 1
+                        if code not in index_messages:
+                            index_messages[code] = message
+                    except Exception as e:
+                        verifier.status(
+                            f"{row[0]} unexpected 'INDEX' error {row[3]}: {str(e)!r}"
+                        )
     click.echo("Operational states:")
     for name, states in operations.items():
         click.echo(f"  {name} states:")
         for state, count in states.items():
             click.echo(f"    {state:>8s} {count:>8d}")
+            if name == "INDEX" and state == "FAILED":
+                for code, count in index_errors.items():
+                    click.echo(
+                        f"           CODE {code:2d}: {count:>4d}  {index_messages[code]}"
+                    )
```

Comment:
FYI: I've been manually checking the various index failure codes, and realized I ought to integrate it here as a special case (since generally those are the only operational errors we see). So:

```
Operational states:
  UPLOAD states:
          OK   117384
  TOOLINDEX states:
       READY   104655
  INDEX states:
          OK   104655
      FAILED      449
           CODE  7:  321  Bad metadata.log file encountered
           CODE  1:  127  Operational error while indexing
           CODE 12:    1  Unexpected error encountered
       READY    12256
```
Getting more detailed information unfortunately requires scrounging in the journal log and/or server-reports indexed documents right now, which is too much to toss together in spare recreational moments... but this is a start.